src/dimension.c

```diff
@@ -478,17 +488,26 @@ dimension_insert_relation(Relation rel, int32 hypertable_id,
 			DirectFunctionCall1(namein, CStringGetDatum(get_func_name(partitioning_func)));
 		values[Anum_dimension_partitioning_func_schema - 1] =
 			DirectFunctionCall1(namein, CStringGetDatum(get_namespace_name(pronamespace)));
+	}
+	else
+	{
+		nulls[Anum_dimension_partitioning_func - 1] = true;
+		nulls[Anum_dimension_partitioning_func_schema - 1] = true;
+	}
+
+	if (num_slices > 0)
+	{
+		/* Closed (hash) dimension */
```

Comment:
Should we assert `interval_length == 0` here? It doesn't look like the caller should supply both a `num_slices` and `interval_length`

Reply:
We could, but not sure this is the right place. The best place to `Assert` this is when we validate the dimension info, although these parameters are checked in `add_dimension` and our schema also protects against both values set. I don't think we need to reassert every time we use them.

Reply:
Actually, changed this to Asserting here, while returning error when validating.